### PR TITLE
Image upload endpoint

### PR DESCRIPTION
**Pull request by SaptakS**
Image uploading to return a temp URL is done in this PR related to #3738 

Sample Input: 
```
{"data": "data:image/gif;base64,R0lGODlhEAAQAMQAAORHHOVSKudfOulrSOp3WOyDZu6QdvCchPGolfO0o/XBs/fNwfjZ0frl3/zy7////wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACH5BAkAABAALAAAAAAQABAAAAVVICSOZGlCQAosJ6mu7fiyZeKqNKToQGDsM8hBADgUXoGAiqhSvp5QAnQKGIgUhwFUYLCVDFCrKUE1lBavAViFIDlTImbKC5Gm2hB0SlBCBMQiB0UjIQA7"}
```

Sample Output:
```
{"url":"/serve_static/media/temp/images/6e35ba8d-0652-489f-9f2f-c68b967bf5f4/Z1JjdnJHeV/1d7965cd-3098-4c81-b532-3cae2185da0b.png"}
```
And the image gets saved in the /static/media/temp/images.. folder.

### CODE REVIEW

**Comment by poush** [comment]
You are saving it to temporary storage inside the project while niranjan comment says to store it in preferred storage.
> Store it in the prefered storage and return the full URL(s) to it as

**Comment by SaptakS** [comment]
Yup. That is the prefered storage for storing the temporary images, I think. @niranjan94 ?

**Comment by poush** [comment]
I think it is actual preferred storage( like s3), especially If you look at the sample endpoints he provided. But still, let @niranjan94 confirm.

**Comment by SaptakS** [comment]
@poush temporary image files were stored in the local storage so that a cron job can delete them. Since the temp files are supposed to be deleted after at max a day. We can do that for s3 or gs as well. But right now it is done in that way for orga-server. Also, if I don't mention a s3 or gs id in the admin panel it's anyway going to store in local system.

**Comment by niranjan94** [comment]
Saptak. Like poush said, by temp I mean preferred storage. Since the url will also be persisted in the database as the original image url in most cases. Do one thing... We actually need both cases. One that temporarily stores in the local file system and one that stores in the preferred storage (not temporary). So have a get parameter like type=temp would store in local temp that is volatile. Anything else, store in preferred storage.

File: app/api/helpers/images.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+from app.api.helpers.storage import UploadedFile
+from flask import current_app
+
+
+def get_image_file_name():
+    return str(uuid.uuid4())
+
+
+def uploaded_image(extension='.png', file_content=None):
+    filename = get_image_file_name() + extension
```
Comment (niranjan94) [inline app/api/helpers/images.py]: Get the file extension from the data string itself.

File: app/api/image_upload.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+upload_routes = Blueprint('upload', __name__, url_prefix='/v1/upload')
+
+
+@jwt_required
+@upload_routes.route('/image', methods=['POST'])
+def upload_image():
+    image = request.json['data']
+    image_file = uploaded_image(file_content=image)
+    force_local request.args.get('force-local', 'false')
```
Comment (niranjan94) [inline app/api/image_upload.py]: call it `force_local`. Let's follow `snake_case` for all GET parameters.

**Pull request closed without merging**